es/13_1_Escribiendo_Puzzle_Scripts.md

```diff
@@ -0,0 +1,523 @@
+Stack: [ ]
+
+Script: OP_2DUP OP_EQUAL OP_NOT OP_VERIFY OP_SHA1 OP_SWAP OP_SHA1 OP_EQUAL
+Stack: [ <numA> <numB> ]
+
+Script: OP_EQUAL OP_NOT OP_VERIFY OP_SHA1 OP_SWAP OP_SHA1 OP_EQUAL
+Running: <numA> <numB> OP_2DUP
+Stack: [ <numA> <numB> <numA> <numB> ]
+```
+
+Luego, nos aseguramos de que los dos números no sean iguales, saliendo si lo son:
+
+```
+Script: OP_NOT OP_VERIFY OP_SHA1 OP_SWAP OP_SHA1 OP_EQUAL
+Running: <numA> <numB> OP_EQUAL
+Stack: [ <numA> <numB> False ]
+
+Script: OP_VERIFY OP_SHA1 OP_SWAP OP_SHA1 OP_EQUAL
+Running: False OP_NOT
+Stack: [ <numA> <numB> True ]
+
+Script: OP_SHA1 OP_SWAP OP_SHA1 OP_EQUAL
+Running: True OP_VERIFY
+Stack: [ <numA> <numB> ] — Does Not Exit
+```
+
+Ahora creamos dos SHA:
+
+```
+Script: OP_SWAP OP_SHA1 OP_EQUAL
+Running: <numB> OP_SHA1
+Stack: [ <numA> <hashB> ]
+
+Script: OP_SHA1 OP_EQUAL
+Running: <numA> <hashB> OP_SWAP
+Stack: [ <hashB> <numA> ]
+
+Script: OP_EQUAL
+Running: <numA> OP_SHA1
+Stack: [ <hashB> <hashA> ]
+```
+
+Finalmente, veremos si coinciden.
+
+```
+Script: 
+Running: <hashB> <hashA> OP_EQUAL
+Stack: [ True ]
+```
+
+Este es un buen script porque muestra un uso cuidadoso de la lógica (con `OP_NOT` y` OP_VERIFY`) y un buen uso de las funciones de pila (con `OP_SWAP`). Es un gran ejemplo de una función del mundo real. Y es del mundo muy real. Cuando [SHA-1 se rompió](https://shattered.io/), 2.48 BTC se liberaron rápidamente de la dirección, con un valor total de aproximadamente $ 3,000 en ese momento.
+
+`btcdeb` se puede ejecutar para probar la colisión (y el script):
+
+```
+$ btcdeb '[255044462d312e330a25e2e3cfd30a0a0a312030206f626a0a3c3c2f57696474682032203020522f4865696768742033203020522f547970652034203020522f537562747970652035203020522f46696c7465722036203020522f436f6c6f7253706163652037203020522f4c656e6774682038203020522f42697473506572436f6d706f6e656e7420383e3e0a73747265616d0affd8fffe00245348412d3120697320646561642121212121852fec092339759c39b1a1c63c4c97e1fffe017f46dc93a6b67e013b029aaa1db2560b45ca67d688c7f84b8c4c791fe02b3df614f86db1690901c56b45c1530afedfb76038e972722fe7ad728f0e4904e046c230570fe9d41398abe12ef5bc942be33542a4802d98b5d70f2a332ec37fac3514e74ddc0f2cc1a874cd0c78305a21566461309789606bd0bf3f98cda8044629a1 255044462d312e330a25e2e3cfd30a0a0a312030206f626a0a3c3c2f57696474682032203020522f4865696768742033203020522f547970652034203020522f537562747970652035203020522f46696c7465722036203020522f436f6c6f7253706163652037203020522f4c656e6774682038203020522f42697473506572436f6d706f6e656e7420383e3e0a73747265616d0affd8fffe00245348412d3120697320646561642121212121852fec092339759c39b1a1c63c4c97e1fffe017346dc9166b67e118f029ab621b2560ff9ca67cca8c7f85ba84c79030c2b3de218f86db3a90901d5df45c14f26fedfb3dc38e96ac22fe7bd728f0e45bce046d23c570feb141398bb552ef5a0a82be331fea48037b8b5d71f0e332edf93ac3500eb4ddc0decc1a864790c782c76215660dd309791d06bd0af3f98cda4bc4629b1 OP_2DUP OP_EQUAL OP_NOT OP_VERIFY OP_SHA1 OP_SWAP OP_SHA1 OP_EQUAL']
```

Comment:
- At the end of command, **replace** `...OP_EQUAL']` **by** `...OP_EQUAL]'`.
- With this change the command will execute correctly in command line.